src/test/java/com/microsoft/aad/msal4j/HttpHeaderTest.java

```diff
@@ -59,4 +69,94 @@ public void testHttpHeaderConstructor_valuesNotSet(){
+        RequestContext requestContext = new RequestContext(
+                app,
+                PublicApi.ACQUIRE_TOKEN_BY_USERNAME_PASSWORD,
+                parameters,
+                UserIdentifier.fromUpn(upn));
+
+        HttpHeaders httpHeaders = new HttpHeaders(requestContext);
+
+        Map<String, String> httpHeaderMap = httpHeaders.getReadonlyHeaderMap();
+
+        String expectedValue = String.format("%s:%s", "upn", upn);
+        Assert.assertEquals(httpHeaderMap.get(HttpHeaders.X_ANCHOR_MAILBOX), expectedValue);
+    }
+
+    @Test
+    public void testHttpHeaderConstructor_userIdentifierHomeAccountId() {
+
+        PublicClientApplication app = PublicClientApplication
+                .builder("client-id")
+                .build();
+
+        IAcquireTokenParameters parameters = UserNamePasswordParameters
+                .builder(Collections.singleton("scopes"), "username", "password".toCharArray())
+                .build();
+
+        String homeAccountId = "userObjectId.userTenantId";
+        RequestContext requestContext = new RequestContext(
+                app,
+                PublicApi.ACQUIRE_TOKEN_SILENTLY,
+                parameters,
+                UserIdentifier.fromHomeAccountId(homeAccountId));
+
+        HttpHeaders httpHeaders = new HttpHeaders(requestContext);
+
+        Map<String, String> httpHeaderMap = httpHeaders.getReadonlyHeaderMap();
+
+        Assert.assertEquals(httpHeaderMap.get(HttpHeaders.X_ANCHOR_MAILBOX), "oid:userObjectId@userTenantId");
+    }
+
+    @Test
+    public void testHttpHeaderConstructor_extraHttpHeadersOverwriteLibraryHeaders() {
+
+        PublicClientApplication app = PublicClientApplication
+                .builder("client-id")
+                .correlationId("correlation-id")
+                .applicationName("app-name")
+                .applicationVersion("app-version")
+                .build();
+
+        String uniqueHeaderKey = "uniqueHeader";
+        String uniqueHeaderValue = "uniqueValue";
+        String uniqueAppName = "my-unique-app-name";
+        Map<String, String> extraHttpHeaders = new HashMap<>();
+        extraHttpHeaders.put(uniqueHeaderKey, uniqueHeaderValue);
+        extraHttpHeaders.put(HttpHeaders.APPLICATION_NAME_HEADER_NAME, uniqueAppName);
+
+        IAcquireTokenParameters parameters = UserNamePasswordParameters
+                .builder(Collections.singleton("scopes"), "username", "password".toCharArray())
+                .extraHttpHeaders(extraHttpHeaders)
+                .build();
+
+        RequestContext requestContext = new RequestContext(
+                app, PublicApi.ACQUIRE_TOKEN_BY_AUTHORIZATION_CODE, parameters);
+
+        HttpHeaders httpHeaders = new HttpHeaders(requestContext);
+
+        Map<String, String> httpHeaderMap = httpHeaders.getReadonlyHeaderMap();
+
+        Assert.assertEquals(httpHeaderMap.get(HttpHeaders.PRODUCT_HEADER_NAME), HttpHeaders.PRODUCT_HEADER_VALUE);
```

Comment:
would be good to have comments to distinguish standard headers, overwritten standard headers and extra header

Reply:
Added some comments.